### PR TITLE
Add failing test fixture for RemoveUnusedVariableAssignRector

### PR DESCRIPTION
**Pull request by Androbin**
# Failing Test for RemoveUnusedVariableAssignRector

Based on https://getrector.org/demo/68325483-921e-4f1e-b290-1b1f252e98f1

### CODE REVIEW

**Comment by Androbin** [comment]
See https://github.com/rectorphp/rector/issues/6494 @TomasVotruba

File: rules-tests/DeadCode/Rector/Assign/RemoveUnusedVariableAssignRector/Fixture/foo.php.inc [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class Foo {
+    public function bar(string $text, bool $flag): string {
+        $text = str_replace('a', 'b', $text);
+        $text = str_replace('c', 'd', $text);
+        if ($flag) {
+            $text = str_replace('d', 'e', $text);
+        }
+        else {
+            $text = str_replace('d', 'f', $text);
+        }
```
Comment (TomasVotruba) [inline rules-tests/DeadCode/Rector/Assign/RemoveUnusedVariableAssignRector/Fixture/foo.php.inc]: The fixture should be as small as possible, to ease the fix writing.

How is this part relevant to the test case? Could it be removed without making test pass?

**Comment by Androbin** [comment]
@TomasVotruba I managed to simplify the test case some more

**Comment by Androbin** [comment]
Removing either branch of the `if/else` or making one run unconditionally doesn't trigger the issue.

**Comment by samsonasik** [comment]
@Androbin I cherry-picked your commit at PR https://github.com/rectorphp/rector-src/pull/205
